src/client/core/evaluator.ts

```diff
@@ -189,8 +189,16 @@ export const evaluate = (
   let variables = { ...inputs }
   const evalOrder = getEvaluationOrder(inputs, constants, operations)
 
-  variables = constants.reduce((vars, { id, value }) => {
-    vars[id] = value
+  variables = constants.reduce((vars, { id, table }) => {
+    // Convert table array to record/object
+    const tableObj = table.reduce((obj, { key, value }) => {
+      obj[key] = value
+      return obj
+    }, <Record<string, number>>{})
+
+    // Evaluate stringified object - the only way to
+    // obtain a non-escaped string of the object.
+    vars[id] = math.evaluate!(JSON.stringify(tableObj))
```

Comment:
```suggestion
    vars[id] = tableObj
```

Just wondering, can we directly set the mapping table this way? https://runkit.com/embed/4jl44k2tcjtv

Reply:
This is a great question. I was playing around with 2 options before settling on the commit:

### Option 1: Setting the value as a Record<string, number>
```
vars[id] = tableObj
```

In order to get this to work, we have to make modifications to several functions to get them to return `Record<string, string | number | Record<string, number>>` instead of `Record<string, string | number>`. 

However, even after doing so, we get this error.
<img width="1071" alt="Screenshot 2021-02-26 at 5 58 49 PM" src="https://user-images.githubusercontent.com/19917616/109285925-9e1da780-785c-11eb-8977-7db55c210fe2.png">

We take a look at the exact code segment for more details on the error. It appears that a possible type of `computed` is `Record<string, ReactText>`.

<img width="957" alt="Screenshot 2021-02-26 at 6 03 07 PM" src="https://user-images.githubusercontent.com/19917616/109286240-f5bc1300-785c-11eb-9670-a2d36eae17f1.png">

And `computed` is defined as the return value of `evaluate()`.

<img width="561" alt="Screenshot 2021-02-26 at 6 04 54 PM" src="https://user-images.githubusercontent.com/19917616/109286416-2a2fcf00-785d-11eb-936f-aecd134f97f2.png">

This happens despite already having coerced the return type of `evaluate()` to be `Record<string, string | number | Record<string, number>>`.

<img width="697" alt="Screenshot 2021-02-26 at 6 02 43 PM" src="https://user-images.githubusercontent.com/19917616/109286272-ff457b00-785c-11eb-89d8-9e1116898f46.png">

This is quite strange and I'm not sure how to fix it. Do let me know if you have any suggestions.


### Option 2: Setting the value as a stringified Record<string, number>
```
vars[id] = JSON.stringify(tableObj)
```

However, this resulted in the variable value being escaped, and so mathjs isn't able to properly handle it. Refer to `T0`.
<img width="485" alt="Screenshot 2021-02-26 at 5 48 47 PM" src="https://user-images.githubusercontent.com/19917616/109284545-f0f65f80-785a-11eb-9236-00135dc5cbe9.png">

### Solution: evaluate a stringified Record<string, number>
```
vars[id] = math.evaluate!(JSON.stringify(tableObj))
```

This produces a value that mathjs is able to use. Refer to `T0`.

<img width="434" alt="Screenshot 2021-02-26 at 5 46 51 PM" src="https://user-images.githubusercontent.com/19917616/109284648-0ec3c480-785b-11eb-84d6-45c008e87299.png">

Reply:
I managed to get it to compile and it seemed to work with the changes [here](https://github.com/opengovsg/checkfirst/compare/feat/add-constants...feat/add-constants-keewei).

Reply:
Merged your proposed changes #189